Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/Concepts.java

```diff
@@ -201,38 +265,74 @@ private JsonValidationResult checkJsonObject(JSONObject jsonObject) {
             return new JsonValidationResult(
-                    "Error parsing request: please provide a concept list for the concept ('conceptlist' attribute).",
+                    "Error parsing request: please provide a description for the concept ('description' attribute).",
                     jsonObject, false);
         }
 
-        String pos = jsonObject.get(JsonFields.POS).toString();
-        if (!POS.posValues.contains(pos)) {
-            logger.error("Error creating concept from REST call. " + pos + " does not exist.");
-            return new JsonValidationResult("POS '" + pos + "' does not exist.", jsonObject, false);
+        // Validation to check if wordnet ids are seperated by comma
+        if (jsonObject.get(JsonFields.WORDNET_ID) != null) {
+            String wordnetIds = jsonObject.get(JsonFields.WORDNET_ID).toString();
+            List<String> wordnetIdsList = Arrays
+                    .asList(wordnetIds.split("\\s*" + Constants.CONCEPT_SEPARATOR + "\\s*"));
+            for (String wordNetId : wordnetIdsList) {
+                try{ 
+                    WordID.parseWordID(wordNetId);
+                } catch (IllegalArgumentException ex) {
+                    return new JsonValidationResult(
+                            "Error parsing request: please provide a valid list of wordnet ids seperated by commas.",
+                            jsonObject, false);
+                }
+            }
         }
 
         return new JsonValidationResult(null, jsonObject, true);
     }
 
-    private ConceptEntry createEntry(JSONObject jsonObject, String username) {
+
+    private ConceptEntry createEntry(JSONObject jsonObject, String username) throws POSMismatchException {
         ConceptEntry conceptEntry = new ConceptEntry();
-        conceptEntry.setCreatorId(username);
-        conceptEntry.setSynonymIds(jsonObject.get(JsonFields.SYNONYM_IDS) != null ? jsonObject.get(
-                JsonFields.SYNONYM_IDS).toString() : "");
-        conceptEntry.setWord(jsonObject.get(JsonFields.WORD).toString());
+        if (jsonObject.get(JsonFields.WORDNET_ID) != null) {
+            conceptEntry.setWordnetId(jsonObject.get(JsonFields.WORDNET_ID).toString());
+            String[] wrappers = jsonObject.get(JsonFields.WORDNET_ID).toString().split(Constants.CONCEPT_SEPARATOR);
+            if (wrappers.length > 0) {
+                ConceptEntry existingConceptEntry = conceptManager.getConceptEntry(wrappers[0]);
+                conceptEntry.setWord(existingConceptEntry.getWord().replace("_", " "));
+                // In case user has entered a POS value. Validate whether POS is
+                // same as wordnet POS
+                if (jsonObject.get(JsonFields.POS) != null) {
+                    if (!existingConceptEntry.getPos().equalsIgnoreCase(jsonObject.get(JsonFields.POS).toString())) {
+                        throw new POSMismatchException("Entered POS " + jsonObject.get(JsonFields.POS).toString()
```

Comment:
I doubt that this would be a huge deal as so far don't really submit many concepts as one, so if you make one or two calls to the db doesn't really affect anything. But if you're worried about that, why don't you just retrieve the wrapped concept (if there is any) before you call validation or creation and just pass it to the to method. This way you would avoid retrieving it twice.